bbq/compiler/desugar.go

```diff
@@ -621,7 +535,16 @@ func (d *Desugar) desugarCondition(condition ast.Condition) ast.Statement {
 }
 
 func (d *Desugar) VisitSpecialFunctionDeclaration(declaration *ast.SpecialFunctionDeclaration) ast.Declaration {
-	return declaration
+	desugaredDecl := d.desugarDeclaration(declaration.FunctionDeclaration).(*ast.FunctionDeclaration)
+	if desugaredDecl == declaration.FunctionDeclaration {
+		return declaration
+	}
+
+	return ast.NewSpecialFunctionDeclaration(
+		d.memoryGauge,
+		declaration.Kind,
+		desugaredDecl,
+	)
```

Comment:
Unrelated fix: Initializers wasn't getting desugared (hence conditions in initializers wasn't working)

Reply:
Good catch! Could you please add a test case for this? (maybe there is one already, I might have missed it)

Reply:
Yeah, there is already a test for this (a test with conditions in the initializer - the desugaring only has an impact if there are conditions)